tools/cldr-apps/src/main/java/org/unicode/cldr/web/UserRegistry.java

```diff
@@ -1056,13 +1060,16 @@ private String updateIntLocs(int id, boolean doCommit, Connection conn, Prepared
         if (user == null) {
             return "";
         }
-
+        logger.finer("uil: remove for user " + id);
         removeIntLoc.setInt(1, id);
         removeIntLoc.executeUpdate();
 
         LocaleSet intLocSet = user.getInterestLocales();
+        logger.finer("uil: intlocs " + id + " = " + intLocSet.toString());
         if (intLocSet != null && !intLocSet.isAllLocales()) {
             for (CLDRLocale loc : intLocSet.getSet()) {
+                logger.finer("uil: intlocs " + id + " + " + loc.toString());
+                updateIntLoc.setInt(1, id);
                 updateIntLoc.setString(2, loc.getLanguage());
```

Comment:
This fixes a bug that I created in https://github.com/unicode-org/cldr/pull/1290 -- thanks for finding it and fixing it!

This code still seems dubious, especially for less-than-managers. For less-than-managers, essentially it appears to be getting the list of "authorized locales" from the cldr_users table, and then copying those locales to the cldr_interest table. For managers-or-stronger, it's getting the list of "interest locales" from the cldr_users table, and then copying those locales to the cldr_interest table. Either way, what's the point of storing the same info twice?

Reply:
@btangmu I think the point was to try to cache the lookup for notifications (with a view to allowing even further customizations, such as a daily digest , etc.) to not require processing every row of the user table and applying policy for each forum post. Since forums are organized by base language ID, there's also a more complex string handling that here is handled via CLDRLocale.

If so, perhaps it should be structured as a temporary table, or even a stored procedure with triggers. But that would move the java logic into SQL, not a great thing for maintainability at this point.

Might be good to DROP the table (if you'll pardon the pun) and just compute the interested users on-the-fly.  this code needs to be readable/maintainable, and ploughing through 2,000 rows is not the most important bottleneck here. (And even if it were, an LRU cache or similar could be used.)

I think the specific issue was just a simple logic regression, and so a simple fix might be good to get notification back up and running. CLDR-14766 proposes restructuring how permissions/etc work to be more maintainable, restructuing interest locales could be part of or adjunct to that.